src/HeatmapLayer.js

```diff
@@ -326,59 +389,83 @@ export default withLeaflet(class HeatmapLayer extends MapLayer {
+    const aggregates = {};
 
-      const p = leafletMap.latLngToContainerPoint(latLng);
+    const accumulateInGrid = (points, leafletMap, bounds, aggregateType) =>
+      reduce(points, (grid, point) => {
+        const latLng = [getLat(point), getLng(point)];
 
-      if (!inBounds(p, bounds)) {
-        return grid;
-      }
+        //skip invalid points
+        if (isInvalidLatLngArray(latLng)) {
+          return grid;
+        }
 
-      const x = Math.floor((p.x - offsetX) / cellSize) + 2;
-      const y = Math.floor((p.y - offsetY) / cellSize) + 2;
+        const p = leafletMap.latLngToContainerPoint(latLng);
 
-      grid[y] = grid[y] || [];
-      const cell = grid[y][x];
+        if (!inBounds(p, bounds)) {
+          return grid;
+        }
 
-      const alt = getIntensity(point);
-      const k = alt * v;
+        const x = Math.floor((p.x - offsetX) / cellSize) + 2;
+        const y = Math.floor((p.y - offsetY) / cellSize) + 2;
 
-      if (!cell) {
-        grid[y][x] = [p.x, p.y, k, 1];
-      } else {
-        cell[0] = (cell[0] * cell[2] + p.x * k) / (cell[2] + k); // x
-        cell[1] = (cell[1] * cell[2] + p.y * k) / (cell[2] + k); // y
-        cell[2] += k; // accumulated intensity value
-        cell[3] += 1;
-      }
+        grid[y] = grid[y] || [];
+        const cell = grid[y][x];
+
+        const key = `${x}-${y}`;
+
+        if (!aggregates[key]) {
+          aggregates[key] = {
+            data: {},
+            same: [],
+            seen: 0
+          };
+        }
 
-      return grid;
-    }, []);
+        aggregates[key].seen++;
+
+        const intensity = getIntensity(point);
+        const agg = computeAggregate(aggregates[key], intensity, aggregateType);
+
+        if (!cell) {
+          grid[y][x] = [p.x, p.y, agg];
+        } else {
+          cell[0] = (cell[0] * cell[2] + p.x * agg) / (cell[2] + agg); // x
+          cell[1] = (cell[1] * cell[2] + p.y * agg) / (cell[2] + agg); // y
+          cell[2] = agg;
+        }
+
+        return grid;
+      },
+      []
+    );
 
     const getBounds = () => new L.Bounds(L.point([-r, -r]), size.add([r, r]));
 
-    const getDataForHeatmap = (points, leafletMap) => roundResults(
-        accumulateInGrid(
-          points,
-          leafletMap,
-          getBounds(leafletMap)
-        )
-      );
+    const getDataForHeatmap = (points, leafletMap, aggregateType) => roundResults(
+      accumulateInGrid(
+        points,
+        leafletMap,
+        getBounds(leafletMap),
+        aggregateType
+      )
+    );
+
+    const data = getDataForHeatmap(
+      this.props.points,
+      this.props.leaflet.map,
+      this.props.aggregateType
+    );
 
-    const data = getDataForHeatmap(this.props.points, this.props.leaflet.map);
+    const totalMax = max(data.map(m => m[2]));
 
     this._heatmap.clear();
-    this._heatmap.data(data).draw(this.getMinOpacity(this.props));
+    this._heatmap.data(data).max(totalMax).draw(this.getMinOpacity(this.props));
```

Comment:
Could we add a `useLocalExtrema` prop, which if set turns on our `totalMax`? This is so that if we contribute this library back, they'll be able to keep using their existing usages as-is.